src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -27,8 +27,23 @@ use errors::Applicability;
 use syntax::ast::{Name, Ident};
 use syntax::attr;
 
-use syntax::ast::{self, Block, ForeignItem, ForeignItemKind, Item, ItemKind, NodeId};
-use syntax::ast::{MetaItemKind, StmtKind, TraitItem, TraitItemKind, Variant};
+use syntax::ast::{
+    self,
+    Block,
+    ExprKind,
+    ForeignItem,
+    ForeignItemKind,
+    Item,
+    ItemKind,
+    MetaItemKind,
+    NodeId,
+    PatKind,
+    StmtKind,
+    TraitItem,
+    TraitItemKind,
+    TyKind,
+    Variant,
+};
```

Comment:
Style nit: please keep single-line imports.
(I found over time that multi-line imports are really interfering with simple grepping / text search.)